crossbeam-channel/src/flavors/array.rs

```diff
@@ -564,6 +534,32 @@ impl<T> Channel<T> {
     }
 }
 
+impl<T: ToOwned<Owned = T>> Channel<T> {
```

Comment:
Ah nice it looks more readable. However, I'd really just use `Clone` instead of `ToOwned<Owned = T>` because I think it's more concise and readable :)

Reply:
It may be, but `Cow`'s trait bound is `ToOwned`; something that implements `ToOwned` does not necessarily implement `Clone`!

https://github.com/rust-lang/rust/blob/d620ae10709ca3669cda89db8b29afcd9accc188/library/alloc/src/borrow.rs#L180-L182

Reply:
Every type that implements `Clone` also implements `ToOwned`: https://doc.rust-lang.org/stable/alloc/borrow/trait.ToOwned.html#impl-ToOwned-1.

Reply:
Yes - but `ToOwned` does not implement `Clone` and the trait bounds of `Cow` use `ToOwned`